src/pybind/mgr/prometheus/module.py

```diff
@@ -1294,15 +1297,15 @@ def get_rbd_stats(self) -> None:
                 namespace_name = s[1]
 
             if pool_name == "*":
-                # collect for all pools
-                osd_map = self.get('osd_map')
-                for pool in osd_map['pools']:
-                    if 'rbd' not in pool.get('application_metadata', {}):
-                        continue
+                # collect for all existing pools
+                for pool in rbd_pools:
                     pool_keys.append((pool['pool_name'], namespace_name))
             else:
-                pool_keys.append((pool_name, namespace_name))
+                for pool in rbd_pools:
+                    if pool['pool_name'] == pool_name:
+                        pool_keys.append((pool_name, namespace_name))  # avoids adding deleted pool
 
+        pool_keys = list(set(pool_keys))
```

Comment:
> Why not make `pool_keys` a set in the first place?

I did try it  pool_keys = list(set([]))`, but it still showed duplicates:
 `[('rbd', None), ('abc', None), ('rbd', None)]`
 I think it's probably bcz when we append it just makes a set and add it to list, so I need to transform it after the loop

Reply:
I'm not following...  What I'm suggesting is replace `pool_keys = []` with `pool_keys = set()` and `pool_keys.append` and `pool_keys.add`.

Reply:
Maybe I can create a tracker for whatever improvement can be done for module? And not block this PR which is unrelated to these code improvements!

Reply:
It's very much related because you are attempting to de-duplicate the user-supplied pool list in *this* PR.

Reply:
> It's very much related because you are attempting to deduplicate the user-supplied pool list in _this_ PR.

Sorry I replied in wrong thread..I meant for the pool_name filtering https://github.com/ceph/ceph/pull/48398#discussion_r993479171

Reply:
The pool_name thing is even more related -- it's a comment on lines of code that this PR is adding...

Reply:
I've applied the suggested changes, nothing breaking! Thanks for suggestions